modules/dnn/src/tensorflow/tf_importer.cpp

```diff
@@ -1113,6 +1115,74 @@ void TFImporter::parseReshape(tensorflow::GraphDef& net, const tensorflow::NodeD
     }
 }
 
+void TFImporter::parseExpandDims(tensorflow::GraphDef& net, const tensorflow::NodeDef& layer, LayerParams& layerParams)
+{
+    const std::string& name = layer.name();
+    const int num_inputs = layer.input_size();
+
+    CV_CheckGT(num_inputs, 0, "");
+    Pin inpId = parsePin(layer.input(0));
+    DataLayout inpLayout = getDataLayout(layer.input(0), data_layouts);
+
+    // Get input shape
+    std::vector<MatShape> inShape_, outShape_;
+    int inpIdindex = layer_id.find(inpId.name)->second;
+
+    dstNet.getLayerShapes(netInputShapes, inpIdindex, inShape_, outShape_);
+    MatShape inpShape = outShape_[0];
```

Comment:
Can you insert an assert, stating that we do not support inpShape.size() != 4?

Reply:
Ok, we can support inpShape.size() == 4 first.

Reply:
Your post states, that it doesn't support anything else. Is that still correct?

Reply:
Hi @rogday, My idea is to let `ExpandDims` support different inputs size, not just 4 dimension. But here are some things I don't understand: 
1. When I use the `DataLayout inpLayout = getDataLayout(layer.input(0), data_layouts);` to get the input data Layout, I found the given `inpLayout` equals to `DATA_LAYOUT_NHWC`. But when I print the size of data, the actual data layout equals `DATA_LAYOUT_NCHW`. Does this mean that tf_importer will automatically convert `NHWC` data into `NCHW` data?

Reply:
inpLayout is unknown when there are no previous nodes with data_layput attribute(otherwise there will be data layout propagation). I think, TF defaults to NHWC. Since OpenCV layers mostly can only operate on NCHW layout, I think it converts only attributes and constants because input is expected to be nchw.

Reply:
@rogday Thank you for your reply, I have updated the code, which could handle 5-dim and mutil-dim input data. And the test data was updated, too. You can found the new test data at [PR of opencv_extra](https://github.com/opencv/opencv_extra/pull/911).